### DISCUSSION TITLE
About Parallel Execution support and Configurable Default Policy

### DISCUSSION THREAD

**Discussion by advaitsoman** [carbon-language/carbon-lang#1731] [Language design]
Hello I liked the idea of Carbon language and its approach of language evolution, 

I was wondering if following point would be useful for language design consideration.

1. Support for parallel execution policy :- 
    Now as hardware becoming more heterogeneous and parallel should carbon language add constructs to effectively represent data structure and function attributes so that they can be easily translated into parallel code. (e.g marking function as thread-safe, parallel loops, support for parallel array transform functions)

2. Support for 'Default Policy' 
   In the language introduction talk one point was raised is now it is difficult to change C++ language defaults (class members are default private) . To avoid such problems in in future for carbon language, can carbon language introduce 'custom global default policy'
All such decision can be programmatically documented in 'default policy' and user should be allowed to make changes in it.
when no explicit policy is specified, compiler can use 'default' 'default policy' (Or perhaps better name :) )

**Comment by geoffromer** [comment]
1. We definitely want to provide excellent support for parallel and concurrent programming, but there hasn't been a lot of in-depth work on that yet.
2. The challenge with this kind of approach will be to allow the language to evolve over time without causing it to fragment into dialects. For that reason, I expect we'll tend to prefer an approach that minimizes the number of user-configurable language settings, and makes it clear that the ability to configure them is temporary.
